megaparse/core/parser/builder.py

```diff
@@ -31,22 +32,28 @@ def build(self, config: ParserConfig) -> MegaParser:
 
 
 class FakeParserBuilder:
-    def build(self, config: ParserConfig) -> MegaParser:
+    def build(self, config: ParserConfig) -> BaseParser:
         """
         Build a fake parser based on the given configuration.
 
         Args:
             config (ParserDict): The configuration to be used for building the parser.
 
         Returns:
-            MegaParser: The built fake parser.
+            BaseParser: The built fake parser.
 
         Raises:
             ValueError: If the configuration is invalid.
         """
 
-        class FakeParser(MegaParser):
-            async def convert(self, file_path: str | Path, **kwargs) -> str:
+        class FakeParser(BaseParser):
+            async def convert(
+                self,
+                file_path: str | Path | None = None,
+                file: IO[bytes] | None = None,
+                **kwargs,
+            ) -> str:
+                print("Fake parser is converting the file")
                 return "Fake conversion result"
 
         return FakeParser()
```

Comment:
what is this used for? for testing? it is weird to have this in the same module as ParserBuilder...